locallib.php

```diff
@@ -1311,69 +1313,35 @@ function moodleoverflow_print_post($post, $discussion, $moodleoverflow, $cm, $co
 
     // Give the option to reply to a post.
     if (moodleoverflow_user_can_post($modulecontext, $post, false)) {
-
         $attributes = [
                 'class' => 'onlyifreviewed',
         ];
-
         // Answer to the parent post.
         if (empty($post->parent)) {
             // Check if limitedanswertime is on.
-            if ($limitedanswertime > time()) {
-                // Change limitedanswertime to a readable date.
-                $date = gmdate('d.m.Y H:i', $limitedanswertime);
-
-                // Check if user is a student or teacher.
-                $roleid = $DB->get_field('role', 'id', ['shortname' => 'editingteacher']);
-                $iseditteacher = $DB->record_exists('role_assignments', ['userid' => $USER->id, 'roleid' => $roleid]);
-
-                $roleidteacher = $DB->get_field('role', 'id', ['shortname' => 'teacher']);
-                $isteacher = $DB->record_exists('role_assignments', ['userid' => $USER->id, 'roleid' => $roleidteacher]);
-                if (!($iseditteacher || $isteacher)) {
-                    // The User is a student.
-
-                    // Build the help icon.
-                    $limitedanswerattributes = ['class' => 'onlyifreviewed text-muted'];
-                    $htmlclass = 'onlyifreviewed text-muted helpicon';
-                    $content = get_string('limitedanswer_helpicon_student', 'moodleoverflow', ['limitedanswerdate' => $date]);
-                    $helpobject = new \mod_moodleoverflow\output\helpicon($htmlclass, $content);
-                    $helpicon = $helpobject->get_helpicon();
-
-                    // Build a html span that has the answer button and the help icon.
-                    $limitedanswerobject = html_writer::tag('span', $str->replyfirst . '    ' . $helpicon);
-
-                    // Save the span in the commands with an extra value.
-                    $commands[] = ['text' => $limitedanswerobject,
-                                        'attributes' => $limitedanswerattributes,
-                                        'limitedanswer' => 'student', ];
-
+            if ($limitedanswersetting->la_starttime != 0 || $limitedanswersetting->la_endtime != 0) {
+                $limitedanswersetting->la_starttime ? $infolimited = " " .
+                    get_string('limitedanswer_info_starttime', 'moodleoverflow',
+                        ['limitedanswerdate' => gmdate('d.m.Y H:i', $limitedanswersetting->la_starttime)]) : $infolimited = '';
+                $limitedanswersetting->la_endtime ? $infolimited .= " " .
+                    get_string('limitedanswer_info_endtime', 'moodleoverflow',
+                        ['limitedanswerdate' => gmdate('d.m.Y H:i', $limitedanswersetting->la_endtime)]) : $infolimited .= '';
+                echo html_writer::div($infolimited, 'alert alert-warning', ['role' => 'alert']);
+            }
+            if (is_currently_time_limited($limitedanswersetting)) {
```

Comment:
Writing another function is not really necessary, as part of the checks in the function were already checked before. Replace the function call with the part of the checks that have not been made so far.